### PR TITLE
Move ATF, u-boot and op-tee OS to 'firmware' folder

### PR DESCRIPTION
**Pull request by rshym**

### CODE REVIEW

**Comment by rshym** [comment]
Tested by build for h3ulcb-4x2g-ab. All related files (bl2, bl31, sa0, sa6, tee, u-boot) are located inside `deploy/images/h3ulcb/firmware` as expected.

File: meta-xt-rcar-driver-domain/inc/firmware_folder.inc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+# Variable FIRMWARE_DEPLOY_DIR_IMAGE is used to specify
```
Comment (lorc) [inline meta-xt-rcar-driver-domain/inc/firmware_folder.inc:2]: For me it looks like this should be `.bbclass`, not `.inc`. Have you considered this?

Comment (rshym) [reply meta-xt-rcar-driver-domain/inc/firmware_folder.inc:2]: Seems like I have found a way to implement this functionality as bbclass. Thanks for pointing.

Comment (rshym) [reply meta-xt-rcar-driver-domain/inc/firmware_folder.inc]: Reworked as `.bbclass`.
Could you please take a look one more time?
Tested, works as expected.

**Review by iusyk — APPROVED** [review]
Reviewed-by: Ihor Usyk <ihor_usyk@epam.com>

File: meta-xt-rcar-driver-domain/classes/collect_firmware.bbclass [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    install -d ${DEPLOY_DIR_IMAGE}/${FIRMWARE_SUBDIR}
+
+    # This task is used after do_deploy, so we can iterate
+    # through DEPLOYDIR to identify files need to be relocated
+    # inside DEPLOY_DIR_IMAGE.
+    for filename in ${DEPLOYDIR}/*; do
+        mv ${DEPLOY_DIR_IMAGE}/`basename ${filename}` ${DEPLOY_DIR_IMAGE}/${FIRMWARE_SUBDIR}
+    done
+}
+addtask do_collect_firmware after do_deploy before do_build
```
Comment (lorc) [inline meta-xt-rcar-driver-domain/classes/collect_firmware.bbclass:27]: Are you sure that it should be `before do_build`?

Comment (rshym) [reply meta-xt-rcar-driver-domain/classes/collect_firmware.bbclass:27]: My understanding is that this line will make `do_build` dependent on `do_collect_firmware`.
Also I goggles a little, and found that same solution  (`before do_build`) is used as way to add task after `do_deploy`.

Comment (lorc) [reply meta-xt-rcar-driver-domain/classes/collect_firmware.bbclass:27]: Ah, yes. I confused `do_compile` and `do_build`. `do_build` is the default task which is "executes" last.

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
Reviewers are aded.

@iusyk could you please merge?

**Pull request merged**